### PR TITLE
Add player sprite to current location

### PR DESCRIPTION
**Pull request by Dragonchitos**
The player sprite from profile indicates the current position. It works for routes, town, dungeon and overlay related to dungeon (like distortion world). 
The green background that indicated the position is removed, so we can see the state of the current location (route or dungeon) more easily.
I didn't test for Alola because I don't have a save in Alola yet.

### CODE REVIEW

File: src/templates/mapOverlay.ejs [hunk 1]
```diff
@@ -25,6 +25,9 @@ Parameters:
     - visible : Required : String
         A knockout data-bind for visibility when true
 
+    - moveToTown : Optional : String
```
Comment (Jaaslet) [inline src/templates/mapOverlay.ejs:28]: Not really a "moveToTown". That kinda indicates a clickaction. Let's call it something like "connectedToTown" or something

File: src/templates/mapDungeon.ejs [hunk 2]
```diff
@@ -15,14 +15,26 @@ Parameters:
     - y : Required : Number
         y axis position of the dungeon icon.
 
+    - notMain : Optional : Boolean
```
Comment (Jaaslet) [inline src/templates/mapDungeon.ejs:18]: Let's call it "isExtension" to match our comments in the SVGs

File: src/components/AlolaSVG.html [hunk 3]
```diff
@@ -1165,7 +1164,7 @@
         <%- include('templates/mapTown',
             { name: "Iki Town"
             , x: 31, y: 16
-            , width: 1, height: 7
+            , width: 1, height: 7, notMain: true
```
Comment (Jaaslet) [inline src/components/AlolaSVG.html:1167]: Let's move all the notMain to a new line. You are renaming them anyways :)

**Review by Jaaslet — CHANGES_REQUESTED** [review]
I LOVE IT! It looks cool. But i still have some comments for you :)
- Some places, i am seeing double
  - MeleMele sea
  - Relic Passage
  - Dark cave
  - Diglett's Cave
- Mega dungeons (like Mt. Chimney) does not show the icon, when you enter it

File: src/styles/variables.less [hunk 4]
```diff
@@ -11,9 +11,8 @@
 
 // Map
 @locked: #000000;
-@currentPlace: #55ff00;
 @incomplete: #ff9100;
-@questAtLocation: #34BF45;
+@questAtLocation: #ff0033;
```
[the remark this answers is not in the document]
Comment (Dragonchitos) [reply src/styles/variables.less:15]: I don't know how to do that. Where should i put the function ?

[the remark this answers is not in the document]
Comment (Jaaslet) [reply src/styles/variables.less:15]: Update.ts

File: src/scripts/worldmap/MapHelper.ts [hunk 5]
```diff
@@ -114,11 +111,21 @@ class MapHelper {
         return cls;
     }
 
-    public static calculateTownCssClass(townName: string): string {
-        // Check if we are currently at this location
+    public static calculateRouteIsCurrentLocation(route: number, region: GameConstants.Region): boolean {
+        if (player.route() == route && player.region == region) {
```
Comment (silverwolf2000) [inline src/scripts/worldmap/MapHelper.ts:115]:
```suggestion
        return (player.route() == route && player.region == region)
```

**Comment by Dragonchitos** [comment]
So : 
 - I did all the rename and move things
 - I add the update function
 - I think I fix Melemele sea, not sure for this one
 - I fix "mega dungeon", which are actually town with several dungeons related without own icon

For Relic Passage, Dark cave and Diglett's Cave : they have two separated icons, so I think it's weird to have the player icon only in one. The player may be confused to click on dungeon and not see the player icon appear, but see it in another place. But I know it's also weird to see two player icons, so I don't know which solution is best.

**Review by Jaaslet — APPROVED** [review]
I have fixed the update function and changed the quest default color to the old current location color.
The rest looks good. Thanks for the PR!

**Comment by Jaaslet** [comment]
> For Relic Passage, Dark cave and Diglett's Cave : they have two separated icons, so I think it's weird to have the player icon only in one. The player may be confused to click on dungeon and not see the player icon appear, but see it in another place. But I know it's also weird to see two player icons, so I don't know which solution is best.

I think two icons is the best solution, so we are keeping that one. We can always change it later, if someone has a good argument.

**Pull request merged**